library/pool.c

```diff
@@ -1184,46 +1197,129 @@ int dnet_io_init(struct dnet_node *n, struct dnet_config *cfg)
 	return err;
 }
 
+int dnet_server_backend_init(struct dnet_node *n, size_t backend_id)
```

Comment:
No, this is not an excuse. Patch with this logic will not go in.
You introduce ugly hacks and bad code for strange and yet unknown reason and the only excuse to add this uglymoron is that if you add it the other way it will be a tiny bit slower? It is much simple just to drop this patch.